### PR TITLE
Fix/authentication passport js

### PR DESCRIPTION
**Pull request by ashley-bergsma**
A few libraries have been added to the project: passportJS, passport-local, passport-google-oauth20 (not being used currently), bcrypt and express-sessions. 

The goal of this commit was to begin the work of offering local login and session support, for users who do not wish to give an email, or use an Oauth login method. Bcrypt has been implemented in the users/register endpoint, and is hashing the user passwords successfully. 

A login route needs to be implemented... the current approach I'm taking isn't working, and I need to research more about bcrypt compareSync before proceeding.

### CODE REVIEW

File: api/routers/auth-router.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+//* all authentication routes (via passport) live here - for the time being Google is being used at the only 'strategy'
+
+//* create an instance of an express router 
+const express = require('express');
+const router = express.Router(); 
+
+// base
```
Comment (ashley-bergsma) [inline api/routers/auth-router.js:7]: This file will probably become obsolete and be removed... it was just a way to get my head around views and passport

File: api/routers/users-model.js [hunk 2]
```diff
@@ -30,3 +31,9 @@ function findById(id) {
         .where({ id })
         .first();
 }
+
+function findByUsername(username) {
```
Comment (ashley-bergsma) [inline api/routers/users-model.js:35]: This function isn't working how I want... I need to research more to understand why

File: api/routers/auth-router.js (hunk 1 above)
Comment (HarrisonMS) [reply api/routers/auth-router.js:7]: sounds good to me hope it helped. Where the views at

File: api/routers/users-model.js (hunk 2 above)
Comment (HarrisonMS) [reply api/routers/users-model.js:35]: that looks like it should query users by username. I think the issue is in the pulling username from params.

**Review by HarrisonMS — APPROVED** [review]

**Pull request merged**